Kickstarter-iOS/Views/Controllers/RootTabBarViewController.swift

```diff
@@ -219,9 +231,16 @@ public final class RootTabBarViewController: UITabBarController {
 }
 
 extension RootTabBarViewController: UITabBarControllerDelegate {
+  public func tabBarController(_ tabBarController: UITabBarController,
```

Comment:
In addition to `didSelect` we had to utilize `shouldSelect` since this is the only place we can check the navigation stack (we simply don't have that information in `didSelect` because it's already reflecting popped state). We need navigation stack in order to prevent from scrolling to top for non-top level controllers.

i.e. scenario:
1. Go to Profile (stack: Profile)
2. Tap Inbox (stack: Profile > Inbox - inbox is on top)
- now tapping scroll to top would scroll Profile while Inbox is being popped of the stack (not desired)